model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -233,12 +237,14 @@ def normalize_slices_attr(node: Node):
                 res_slices.append(slice(s, s + 1, strides[i]))  # need strides if shrink index is negative
             elif node.ellipsis_mask[i]:
                 for idx in range(num_ellipsis_inserts):
-                    res_slices.append(slice(0, data_shape[in_idx], 1))
+                    res_slices.append(slice(0, None, 1))  # slice till the end of the dimension
```

Comment:
The idea of this change was to properly support cases when input dimension is dynamic. The Python slice semantic supports output shape calculation using `None` in the `stop` value out of the box. If the input shape is dynamic then the value of `data_shape[in_idx]` will be `dynamic_dimension` and we need to handle that in the places where `slices` attribute is used. In particular in the `ConvertGroupedStridedSlice` transformation, `get_shape_from_slice` function. I have checked that I have already added such kind of handling there, but I will run an additional validation to make sure that I have not broken something else. Anyway, the `ConvertGroupedStridedSlice` will be much limited in terms of when it can be applied since the `data_shape[in_idx]` may be `dynamic_dimension`.